lib/topic_view.rb

```diff
@@ -735,6 +721,67 @@ def read_posts_set
 
   private
 
+  def load_mentioned_users
+    mentions = load_mentions
+    users = load_users(mentions.values.flatten.uniq)
+
+    @mentioned_users = mentions.map do |post_id, user_ids|
+      mentioned_users = user_ids.map { |id| users[id] }.compact
+      [post_id, mentioned_users]
+    end.to_h
+  end
+
+  def load_users(ids)
+    users = User.where(id: ids)
+    users = users.includes(:user_status) if SiteSetting.enable_user_status
+    users.to_h { |u| [u.id, u] }
+  end
+
+  def load_mentions
```

Comment:
I wonder if this should be `Post.lookup_mentions(posts)` instead. This would allow us to eliminate `Post#mentions` and just have one method which is responsible for looking up and setting the cache.

Reply:
As I explained in discussion above, we can't just move all code to `TopicView`.

If we move the code, what would we do when we have only a `post` model at hand and don't have a `TopicView` instance?

Reply:
Just call Post.lookup_mentions and pass it a single post? As long as the method can lookup mentions for an array of posts. It doesn’t matter whether we have one post or many posts.

Reply:
Oh, sorry, I see now, that should work 👍